src/sys.zig

```diff
@@ -680,6 +643,103 @@ pub fn ntCreateFile(
     }
 }
 
+pub fn openFileAtWindowsT(
```

Comment:
this is so much nicer.
something i think about and it's fine to not do this now consideirng the size of the pr,

is maybe `access_mask`, `disposition`, etc can be shoved into an options struct similar to what std.os.windows.OpenFile does. there we can hvae the common defaults and also expose other flags (like the share args, but i dont think we'll actually need to customize that part)

Reply:
yeah I think that would be a good idea. in general I think working with options structs is nice for these more complicated APIs